packages/app-project/src/shared/components/SubjectPicker/helpers/searchParams.js

```diff
@@ -1,10 +1,10 @@
 export default function searchParams(data) {
-  let query = ''
+  let query = []
   Object.entries(data).forEach(([key, value]) => {
     if (value !== '') {
-      query += `@${key}:${value}*`
+      query.push(`${key}__contains=${value}`)
```

Comment:
I'd take that suggestion a step further and suggest `dompurify` (I have no idea what datasette does to sanitize incoming parameters).

Reply:
Datasette uses parameterised SQL queries, which should automatically protect against injection attacks. 

I’m not sure if dompurify can be used to clean up URL queries but I’ll have a look. It might just protect against XSS in HTML.

Reply:
I checked and dompurify is a sanitiser for HTML, MathML and SVG strings.

We might need URL-encoding if someone wants to search for reserved characters like &, /, . or ? (eg. search for this string `some text&rowid=2`.) I'll play around and see if Grommet DataTable passes the value as-is.

Reply:
`DataTable` doesn't encode the text input. I'm able to select a single row from the database by typing `jan&rowid=2` into the Date search box, which sends this query: https://subject-set-search-api.zooniverse.org/subjects-4dec399/92753.json?Date__contains=jan&rowid=2&_sort=priority

Reply:
That URL issues this SQL query:
```sql
select rowid, priority, subject_id, Page, Date from [92753] where \"Date\" like :p0 and \"rowid\" = :p1 order by priority limit 101
```

where the params are `p0: "jan"` and `p1: 2`.

Reply:
I've fixed this and updated the test case to search for `Smith & Jones` instead of `Smith`.

Reply:
@camallen pointed out that we also run Datasette in immutable mode, which gives us an extra layer of defence. The data stored in those tables can't be changed without rebuilding the service.

https://subject-set-search-api.zooniverse.org/-/databases

https://docs.datasette.io/en/latest/performance.html?highlight=immutable#immutable-mode